src/mqt/bench/output.py

```diff
@@ -86,8 +85,10 @@ def generate_header(
         f"// Output format: {fmt.value}",
     ))
 
-    if gateset:
+    if target:
+        gateset = sorted({str(inst.name) for inst, _ in target.instructions})
         lines.append(f"// Used gateset: {gateset}")
+    c_map = target.build_coupling_map() if target else None
```

Comment:
Pretty sure this function will need an additional `level` input.
Because otherwise, this will add a coupling map even if a circuit is only transpiled to the native gates level

Reply:
I don't think so, since the `build_coupling_map()` functions returns `None` according to its docstring if there are no connectivity constraints present in the target (see https://github.com/Qiskit/qiskit/blob/ee4095933e688efb794e679b3feda658fefdec9b/qiskit/transpiler/target.py#L545).